### ISSUE TITLE
Test failure (numerical noise?)

### ISSUE THREAD

**Issue by vbraun** [Normaliz/Normaliz#340]
I'm getting this on Debian 9 x86_64. Looks like numerical noise to me, e.g. 1.236068 vs 1.2360680
~~~~
[normaliz-3.7.2] diff -b test-nf/dodecahedron-v.ref test-nf/dodecahedron-v.out | tee test-nf/dodecahedron-v.diff
[normaliz-3.7.2] 35,54c35,54
[normaliz-3.7.2] <    (-a+1 ~ -1.236068) (-2*a+4 ~ -0.472136)                    0 1
[normaliz-3.7.2] <    (-a+1 ~ -1.236068)   (2*a-4 ~ 0.472136)                    0 1
[normaliz-3.7.2] <     (a-3 ~ -0.763932)    (a-3 ~ -0.763932)    (a-3 ~ -0.763932) 1
[normaliz-3.7.2] <     (a-3 ~ -0.763932)    (a-3 ~ -0.763932)    (-a+3 ~ 0.763932) 1
[normaliz-3.7.2] <     (a-3 ~ -0.763932)    (-a+3 ~ 0.763932)    (a-3 ~ -0.763932) 1
[normaliz-3.7.2] <     (a-3 ~ -0.763932)    (-a+3 ~ 0.763932)    (-a+3 ~ 0.763932) 1
[normaliz-3.7.2] <  (-2*a+4 ~ -0.472136)                    0   (-a+1 ~ -1.236068) 1
[normaliz-3.7.2] <  (-2*a+4 ~ -0.472136)                    0     (a-1 ~ 1.236068) 1
[normaliz-3.7.2] <                     0   (-a+1 ~ -1.236068) (-2*a+4 ~ -0.472136) 1
[normaliz-3.7.2] <                     0   (-a+1 ~ -1.236068)   (2*a-4 ~ 0.472136) 1
[normaliz-3.7.2] <                     0     (a-1 ~ 1.236068) (-2*a+4 ~ -0.472136) 1
[normaliz-3.7.2] <                     0     (a-1 ~ 1.236068)   (2*a-4 ~ 0.472136) 1
[normaliz-3.7.2] <    (2*a-4 ~ 0.472136)                    0   (-a+1 ~ -1.236068) 1
[normaliz-3.7.2] <    (2*a-4 ~ 0.472136)                    0     (a-1 ~ 1.236068) 1
[normaliz-3.7.2] <     (-a+3 ~ 0.763932)    (a-3 ~ -0.763932)    (a-3 ~ -0.763932) 1
[normaliz-3.7.2] <     (-a+3 ~ 0.763932)    (a-3 ~ -0.763932)    (-a+3 ~ 0.763932) 1
[normaliz-3.7.2] <     (-a+3 ~ 0.763932)    (-a+3 ~ 0.763932)    (a-3 ~ -0.763932) 1
[normaliz-3.7.2] <     (-a+3 ~ 0.763932)    (-a+3 ~ 0.763932)    (-a+3 ~ 0.763932) 1
[normaliz-3.7.2] <      (a-1 ~ 1.236068) (-2*a+4 ~ -0.472136)                    0 1
[normaliz-3.7.2] <      (a-1 ~ 1.236068)   (2*a-4 ~ 0.472136)                    0 1
[normaliz-3.7.2] ---
[normaliz-3.7.2] >     (-a+1 ~ -1.2360680) (-2*a+4 ~ -0.47213595)                      0 1
[normaliz-3.7.2] >     (-a+1 ~ -1.2360680)   (2*a-4 ~ 0.47213595)                      0 1
[normaliz-3.7.2] >     (a-3 ~ -0.76393202)    (a-3 ~ -0.76393202)    (a-3 ~ -0.76393202) 1
[normaliz-3.7.2] >     (a-3 ~ -0.76393202)    (a-3 ~ -0.76393202)    (-a+3 ~ 0.76393202) 1
[normaliz-3.7.2] >     (a-3 ~ -0.76393202)    (-a+3 ~ 0.76393202)    (a-3 ~ -0.76393202) 1
[normaliz-3.7.2] >     (a-3 ~ -0.76393202)    (-a+3 ~ 0.76393202)    (-a+3 ~ 0.76393202) 1
[normaliz-3.7.2] >  (-2*a+4 ~ -0.47213595)                      0    (-a+1 ~ -1.2360680) 1
[normaliz-3.7.2] >  (-2*a+4 ~ -0.47213595)                      0      (a-1 ~ 1.2360680) 1
[normaliz-3.7.2] >                       0    (-a+1 ~ -1.2360680) (-2*a+4 ~ -0.47213595) 1
[normaliz-3.7.2] >                       0    (-a+1 ~ -1.2360680)   (2*a-4 ~ 0.47213595) 1
[normaliz-3.7.2] >                       0      (a-1 ~ 1.2360680) (-2*a+4 ~ -0.47213595) 1
[normaliz-3.7.2] >                       0      (a-1 ~ 1.2360680)   (2*a-4 ~ 0.47213595) 1
[normaliz-3.7.2] >    (2*a-4 ~ 0.47213595)                      0    (-a+1 ~ -1.2360680) 1
[normaliz-3.7.2] >    (2*a-4 ~ 0.47213595)                      0      (a-1 ~ 1.2360680) 1
[normaliz-3.7.2] >     (-a+3 ~ 0.76393202)    (a-3 ~ -0.76393202)    (a-3 ~ -0.76393202) 1
[normaliz-3.7.2] >     (-a+3 ~ 0.76393202)    (a-3 ~ -0.76393202)    (-a+3 ~ 0.76393202) 1
[normaliz-3.7.2] >     (-a+3 ~ 0.76393202)    (-a+3 ~ 0.76393202)    (a-3 ~ -0.76393202) 1
[normaliz-3.7.2] >     (-a+3 ~ 0.76393202)    (-a+3 ~ 0.76393202)    (-a+3 ~ 0.76393202) 1
[normaliz-3.7.2] >       (a-1 ~ 1.2360680) (-2*a+4 ~ -0.47213595)                      0 1
[normaliz-3.7.2] >       (a-1 ~ 1.2360680)   (2*a-4 ~ 0.47213595)                      0 1
[normaliz-3.7.2] 59,70c59,70
[normaliz-3.7.2] <  (-1/4*a-1/4 ~ -0.809017)                        0       (-1/2 ~ -0.500000) 1
[normaliz-3.7.2] <  (-1/4*a-1/4 ~ -0.809017)                        0         (1/2 ~ 0.500000) 1
[normaliz-3.7.2] <        (-1/2 ~ -0.500000) (-1/4*a-1/4 ~ -0.809017)                        0 1
[normaliz-3.7.2] <        (-1/2 ~ -0.500000)   (1/4*a+1/4 ~ 0.809017)                        0 1
[normaliz-3.7.2] <                         0       (-1/2 ~ -0.500000) (-1/4*a-1/4 ~ -0.809017) 1
[normaliz-3.7.2] <                         0       (-1/2 ~ -0.500000)   (1/4*a+1/4 ~ 0.809017) 1
[normaliz-3.7.2] <                         0         (1/2 ~ 0.500000) (-1/4*a-1/4 ~ -0.809017) 1
[normaliz-3.7.2] <                         0         (1/2 ~ 0.500000)   (1/4*a+1/4 ~ 0.809017) 1
[normaliz-3.7.2] <          (1/2 ~ 0.500000) (-1/4*a-1/4 ~ -0.809017)                        0 1
[normaliz-3.7.2] <          (1/2 ~ 0.500000)   (1/4*a+1/4 ~ 0.809017)                        0 1
[normaliz-3.7.2] <    (1/4*a+1/4 ~ 0.809017)                        0       (-1/2 ~ -0.500000) 1
[normaliz-3.7.2] <    (1/4*a+1/4 ~ 0.809017)                        0         (1/2 ~ 0.500000) 1
[normaliz-3.7.2] ---
[normaliz-3.7.2] >  (-1/4*a-1/4 ~ -0.80901699)                          0       (-1/2 ~ -0.50000000) 1
[normaliz-3.7.2] >  (-1/4*a-1/4 ~ -0.80901699)                          0         (1/2 ~ 0.50000000) 1
[normaliz-3.7.2] >        (-1/2 ~ -0.50000000) (-1/4*a-1/4 ~ -0.80901699)                          0 1
[normaliz-3.7.2] >        (-1/2 ~ -0.50000000)   (1/4*a+1/4 ~ 0.80901699)                          0 1
[normaliz-3.7.2] >                           0       (-1/2 ~ -0.50000000) (-1/4*a-1/4 ~ -0.80901699) 1
[normaliz-3.7.2] >                           0       (-1/2 ~ -0.50000000)   (1/4*a+1/4 ~ 0.80901699) 1
[normaliz-3.7.2] >                           0         (1/2 ~ 0.50000000) (-1/4*a-1/4 ~ -0.80901699) 1
[normaliz-3.7.2] >                           0         (1/2 ~ 0.50000000)   (1/4*a+1/4 ~ 0.80901699) 1
[normaliz-3.7.2] >          (1/2 ~ 0.50000000) (-1/4*a-1/4 ~ -0.80901699)                          0 1
[normaliz-3.7.2] >          (1/2 ~ 0.50000000)   (1/4*a+1/4 ~ 0.80901699)                          0 1
[normaliz-3.7.2] >    (1/4*a+1/4 ~ 0.80901699)                          0       (-1/2 ~ -0.50000000) 1
[normaliz-3.7.2] >    (1/4*a+1/4 ~ 0.80901699)                          0         (1/2 ~ 0.50000000) 1
[normaliz-3.7.2] diff -b test-nf/dodecahedron-v.ref test-nf/dodecahedron-v.out > test-nf/dodecahedron-v.diff
[normaliz-3.7.2] /var/lib/buildbot/slave/sage_git/build/local/var/tmp/sage/build/normaliz-3.7.2/src/test/Makefile.classic:76: recipe for target 'test-nf/dodecahedron-v.diff' failed
[normaliz-3.7.2] make[5]: *** [test-nf/dodecahedron-v.diff] Error 1
[normaliz-3.7.2] make[5]: Leaving directory '/var/lib/buildbot/slave/sage_git/build/local/var/tmp/sage/build/normaliz-3.7.2/src/test/run_tests'
[normaliz-3.7.2] Makefile:1001: recipe for target 'check-local' failed
[normaliz-3.7.2] make[4]: *** [check-local] Error 2
[normaliz-3.7.2] make[4]: Leaving directory '/var/lib/buildbot/slave/sage_git/build/local/var/tmp/sage/build/normaliz-3.7.2/src/test'
[normaliz-3.7.2] Makefile:882: recipe for target 'check-am' failed
[normaliz-3.7.2] make[3]: *** [check-am] Error 2
[normaliz-3.7.2] make[3]: Leaving directory '/var/lib/buildbot/slave/sage_git/build/local/var/tmp/sage/build/normaliz-3.7.2/src/test'
[normaliz-3.7.2] Makefile:427: recipe for target 'check-recursive' failed
[normaliz-3.7.2] make[2]: *** [check-recursive] Error 1
[normaliz-3.7.2] make[2]: Leaving directory '/var/lib/buildbot/slave/sage_git/build/local/var/tmp/sage/build/normaliz-3.7.2/src'
[normaliz-3.7.2] 
[normaliz-3.7.2] real	0m38.969s
[normaliz-3.7.2] user	1m47.568s
[normaliz-3.7.2] sys	0m4.172s
[normaliz-3.7.2] ************************************************************************
[normaliz-3.7.2] Error testing package normaliz-3.7.2
[normaliz-3.7.2] ************************************************************************
~~~~

**Comment by w-bruns** [comment]
The output is generated by the operator << in the e-antic C++ wrapper. It seems that the decimal precision of the rational approximations has been increased by 2 digits from 6 to 8 in the output above.

**Comment by vbraun** [comment]
And I just upgrade to e-antic 0.1.7, that must be it

**Comment by fingolfin** [comment]
Perhaps we can/should (re)set the display precision? And/or add an option so the user can tweak it, then use that in the test suite?

**Comment by isuruf** [comment]
Any updates on this?

**Comment by w-bruns** [comment]
Will be taken care of by the next release(hopefully within the next week). The solution is to omit the critical floating point  output in the comparison of the files. Unfortunately it is not possible to choose the number of digits.

**Comment by w-bruns** [comment]
I have now realized a test in master that skips the floating point approximations of algebaric numbers. See test/Makefile.classic.

Disadvantage: the method produces auxiliary files that are not aitomatically deleted. Suggestions for improvement are very welcome.

**Comment by w-bruns** [comment]
Solved in 3.8.7.
